service/common/src/main/java/org/apache/polaris/service/context/RealmContextResolver.java

```diff
@@ -19,22 +19,45 @@
 package org.apache.polaris.service.context;
 
 import java.util.Map;
+import java.util.concurrent.CompletionStage;
 import java.util.function.Function;
 import org.apache.commons.collections.map.CaseInsensitiveMap;
 import org.apache.polaris.core.context.RealmContext;
 
+/**
+ * An interface for resolving the realm context for a given request.
+ *
+ * <p>General implementation guidance:
+ *
+ * <p>Methods in this class should not block the calling thread. If the realm resolution process is
+ * blocking, it should be done in a separate thread.
+ *
+ * <p>In the case of an error during realm resolution, the {@link CompletionStage} should complete
+ * exceptionally; methods should not throw exceptions directly.
```

Comment:
[doubt] can you please elaborate on, who throws the exception ?

Reply:
@singhpk234 as usual for asynchronous code, invoking `resolveRealmContext()` should never fail. If a specific implementation is unable to resolve the realm for whatever reason, it should return a failed future.